### DISCUSSION TITLE
Show the slow mode timer as a hint in the chat input

### DISCUSSION THREAD

**Discussion by x9136** [Chatterino/chatterino2#5425] [Ideas] [answered]
I suggest adding a slow mode timer (and possibly other timers, such as timeout) to the chat input, so as not to send messages for the sake of checking the timer. Counting down the timer in your head, of course, is much more inconvenient than even sending several messages to check it. For example:

**(slow 10s) Send message as username...**
↓
**(slow 9s) Send message as username...**
↓
...
↓
**Send message as username...**

**How to track it?**
After sending a message in the slow mod, there is now a response like: "This room is in slow mode and you are sending messages too quickly. You will be able to talk again in 12 seconds." Since we know from now on that we have 12 seconds to wait, we can immediately send this info as a hint in the chat input field. Other timers can also be tracked after you have sent a message for check.

It is best if this info is taken earlier: at the moment when the message that led to the restriction was sent or when some kind of restriction appeared, as in the case of the timeout. But I have a feeling that there are issues with this, so I suggest an option a step later.

**Comment by Felanbird** [comment, marked as answer]
https://github.com/Chatterino/chatterino2/pull/3329#issuecomment-2002417296

**Reply by Felanbird** [reply]
slow mode specifically, we can't 100% of the time correctly indicate the timer if the channel is/isn't allowing subs to bypass slow mode, also messages that are dropped by Twitch we also can't correctly indicate whether the timer should start
